kong/runloop/handler.lua

```diff
@@ -1433,7 +1433,8 @@ return {
 
       local upstream_status_header = constants.HEADERS.UPSTREAM_STATUS
       if kong.configuration.enabled_headers[upstream_status_header] then
-        header[upstream_status_header] = tonumber(sub(var.upstream_status or "", -3))
+        local upstream_status = ctx.buffered_status or tonumber(sub(var.upstream_status or "", -3))
+        header[upstream_status_header] = tostring(upstream_status)
```

Comment:
I see you added `tostring` here. Not sure why. Perhaps then the `tonumber` above is unwanted. That is now it can set header to `"nil"`. Not sure can it happen but still. Perhaps just remove the `tostring`?

Reply:
Removing. I was trying to eliminate a warning.